provision/README.md

```diff
@@ -1,43 +1,142 @@
+| MF_PROVISION_PASS                   | Mainflux user password                            | password                       |
+| MF_PROVISION_BS_SVC_URL             | Mainflux Bootstrap service URL                    | http://localhost/things/configs |
+| MF_PROVISION_BS_SVC_WHITELIST_URL   | Mainflux Bootstrap service whitelist URL          | http://localhost/things/state  |
+| MF_PROVISION_CERTS_SVC_URL          | Certificats service URL                           | http://localhost/certs         |
+| MF_PROVISION_X509_PROVISIONING      | Should X509 client cert be provisioned            | false                          |
+| MF_PROVISION_BS_CONFIG_PROVISIONING | Should thing config be saved in Bootstrap service | true                           |
+| MF_PROVISION_BS_AUTO_WHITEIST       | Should thing be auto whitelisted                  | true                           |
+| MF_PROVISION_BS_CONTENT             | Bootstrap service content                         | {}                             |
+
+By default, call to `/mapping` endpoint will create one thing and two channels (`control` and `data`) and connect it. If there is a requirement for different provision layout we can use [config](docker/configs/config.toml) file in addition to environment variables. For the purposes of running provision as an add-on in docker composition environment variables seems more suitable. Environoment variables are set in [.env](.env).  
+Configuration can be specified in [config.toml](configs/config.toml). Config file can specify all the settings that environment variables can configure and in addition
+`/mapping` endpoint provision layout can be configured.
+
+In `config.toml` we can enlist array of things and channels that we want to create and make connections between them which we call provision layout.
+Metadata can be whatever suits your needs except that at least one thing needs to have `external_id` (which is populated with value from [request](#example)).
+For channels metadata `type` is reserved for `control` and `data` which we use with [Agent](https://github.com/mainflux/agent).
+
+Example below
+```
+[[things]]
+  name = "thing"
+
+  [things.metadata]
+    external_id = "xxxxxx"
+
+
+[[channels]]
+  name = "control-channel"
+
+  [channels.metadata]
+    type = "control"
+
+[[channels]]
+  name = "data-channel"
+
+  [channels.metadata]
+    type = "data"
+
+[[channels]]
+  name = "export-channel"
+
+  [channels.metadata]
+    type = "data"
+```
+
+## Authentication
+In order to create necessary entities provision service needs to authenticate against Mainflux API. To provide authentication credentials to the provision service you can pass it in an environment variable or in a config file as Mainflux user and password or as API token (that can be issued on users/keys endpoint). Additionally users or API token can be passed in Authorization header, this authentication takes precedence over others.
```

Comment:
Is this "users/keys" path? If it is, use backsticks. If it's not, maybe explain better where API token can be issued.